components/base/wheeled/wheeled_base.go

```diff
@@ -130,64 +130,53 @@ func (wb *wheeledBase) Reconfigure(ctx context.Context, deps resource.Dependenci
 
 	wb.spinSlipFactor = newConf.SpinSlipFactor
 
-	// Check if wb.left is different from newConf.Left before changing wb.left
-	if len(wb.left) != len(newConf.Left) {
-		// Resetting the left motor list
-		wb.left = make([]motor.Motor, 0)
-
-		for _, name := range newConf.Left {
-			m, err := motor.FromDependencies(deps, name)
-			if err != nil {
-				return errors.Wrapf(err, "no left motor named (%s)", name)
+	updateMotors := func(curr []motor.Motor, fromConfig []string) ([]motor.Motor, error) {
+		newMotors := make([]motor.Motor, 0)
+		if len(curr) != len(fromConfig) {
+			for _, name := range fromConfig {
+				select {
+				case <-ctx.Done():
+					return newMotors, resource.NewBuildTimeoutError(wb.Name())
+				default:
+				}
+				m, err := motor.FromDependencies(deps, name)
+				if err != nil {
+					return newMotors, errors.Wrapf(err, "no motor named (%s)", name)
+				}
+				newMotors = append(newMotors, m)
 			}
-			wb.left = append(wb.left, m)
-		}
-	} else {
-		// Compare each element of the slices
-		for i := range wb.left {
-			if wb.left[i].Name().String() != newConf.Left[i] {
-				// Resetting the left motor list
-				wb.left = make([]motor.Motor, 0)
-
-				for _, name := range newConf.Left {
-					m, err := motor.FromDependencies(deps, name)
-					if err != nil {
-						return errors.Wrapf(err, "no left motor named (%s)", name)
+		} else {
+			// Compare each element of the slices
+			for i := range curr {
+				select {
+				case <-ctx.Done():
+					return newMotors, resource.NewBuildTimeoutError(wb.Name())
+				default:
+				}
+				if (curr)[i].Name().String() != (fromConfig)[i] {
+					for _, name := range fromConfig {
+						m, err := motor.FromDependencies(deps, name)
+						if err != nil {
+							return newMotors, errors.Wrapf(err, "no motor named (%s)", name)
+						}
+						newMotors = append(newMotors, m)
 					}
-					wb.left = append(wb.left, m)
+					break
 				}
-				break
 			}
 		}
+		return newMotors, nil
 	}
 
-	if len(wb.right) != len(newConf.Right) {
```

Comment:
this check is now missing? This is physically  important for wheeled bases phsycially

Reply:
A differential drive relies on an equal number of motors on the left and right side to supply power to the motors to drive it. This breaks all our math if it isn't true.